repaygithub/ui-tools
# Cactus 713 q3

# Overview

[CACTUS-713](https://repayonline.atlassian.net/browse/CACTUS-713)

**Removed dependencies** on `@repay/eslint-config`,  `@repay/testing-tools`. **Added missing dependencies** on `@repay/scripts`. 
For that, I found a tool called [depcheck](https://www.npmjs.com/package/depcheck).

### Upgrading deps: 
Two dependencies had Patch Updates 🟢 . 
Almost 5 dependencies were updated with Minor Updates 🟡 . 
Regarding Major Updates 🔴 , almost 9 packages were updated, their breaking changes are mostly related to `esm` implementations. `webpack-dev-server` had the most important breaking changes.  

## Testing 

### babel-preset and repay-scripts

1. `cd modules/babel-preset` and `yalc publish`
2. `cd ../repay-scripts` and `yalc publish`
(or, `cd modules/babel-preset && yalc publish && cd ../repay-scripts && yalc publish` 😄 )
3. Go to the Cactus repo and run `yalc add @repay/babel-preset && yalc add @repay/scripts`
4. Run `yarn install`
5. Run `yarn cleanup` (it will error because babel-preset and repay-scripts don't have `cleanup` commands, but that's okay), followed by `yarn build`.
6.  `cd examples/mock-ebpp` and `yalc add @repay/scripts`.
7. Run `yarn install`. (this step and the 6th one is just for being sure that the `mock-ebpp` is adding our updated dependencies correctly) 
6. Run `yarn start`.  Make sure the mock EBPP app runs fine.
7. Test the mock-ebpp app on E11 too! 

### eslint-config 

1. `cd modules/eslint-config` and `yalc publish`
2. Go to the Cactus repo and run `yalc add @repay/eslint-config`
4. Run `yarn install`
5. Run `yarn lint` and make sure things look okay.

### testing-tools
1. `cd modules/testing-tools` and `yalc publish`
2. Go to the cactus repo and `cd examples/mock-ebpp/` and `yalc add @repay/testing-tools`
3. Run `yarn install`
4. Run `yarn test && yarn test:local` and make sure all the test pass. 


### Notes 

#### .testcaferc.json

Due to the following warning on the current `testcafe` version: 

```bash
The 'tsConfigPath' option is deprecated and will be removed in the next major release. Use the 'compilerOptions.typescript.configPath' option instead.
 ```

The `tsConfigPath` present on every `.testcaferc.json` file must be replaced by: 
```json
"compilerOptions": {
    "typescript":{
      "configPath": "tsconfig.json"
    }
  }
```

#### Test description on `cactus/modules/cactus-theme/tests/exports.test.ts`
😅  Not sure if this needs a ticket but the test description says `@repay/cactus-i18n` instead of `@repay/cactus-theme`


#### Tempy and ESM
As I commented on [this thread](https://repay.slack.com/archives/CABGTU4S3/p1634327114041300?thread_ts=1634255481.037500&cid=CABGTU4S3) , We're getting a parsing error in the jest library. Probably the solution involves updating the syntax of every JS file, as Dhalton said. So We need to bear in mind this for a new ticket, I guess.
